client/src/language/diagnosticsSupport.ts

```diff
@@ -116,6 +119,22 @@ const checkHasSupportedSource = (diagnostic: vscode.Diagnostic): boolean => {
   )
 }
 
+const checkIsIgnoredPylanceUndefinedVariable = async (
+  diagnostic: vscode.Diagnostic,
+  originalTextDocument: vscode.TextDocument,
+  adjustedRange: vscode.Range
+): Promise<boolean> => {
+  if (diagnostic.source?.includes('Pylance') !== true) {
+    return false
+  }
+  if (typeof diagnostic.code !== 'object' || diagnostic.code?.value !== 'reportUndefinedVariable') {
+    return false
+  }
+
+  const definition = await requestsManager.getDefinition(originalTextDocument, adjustedRange.end)
```

Comment:
Based on my personal experience in other languages, that sort of functionality adds a lot to the user experience. It adds confidence the code won't crash before being run, and saves a lot of time on such issues.

On my side the diagnostics take some delay to update, but to me it seems comparable to other extensions. Maybe optimizations could be done somewhere else when the document is analyzed? Maybe we could save some power by adding debouncing so it does not update the diagnostics while the user is typing?

The main issue I see is that in our case it is not 100% reliable. For example, on `meta/classes-recipe/cml1.bbclass`, it still gives a diagnostic saying `src_patches` is not defined. I admit I don't understand BitBake's rule there, since I don't see it being imported anywhere.

Reply:
If performance is alright then it's better to get accuracy indeed.

I also thought about grouping individual symbols requests into one containing all definitions for all symbols, and caching it. It was required to do that for a lot of the bitbake context which was huge and slow to get. I don't think it's necessary here since onDefinition is a very singular request. Ok as is.